src/framework/interceptor/wrap.clj

```diff
@@ -0,0 +1,31 @@
+(ns framework.interceptor.wrap
+  (:require
+   [xiana.core :as xiana]))
+
+(defn interceptor
+  "Interceptor wrapper to use xiana monad."
+  [in]
+  (cond-> {}
+    (:enter in) (assoc :enter (fn [state] (xiana/ok ((:enter in) state))))
+    (:leave in) (assoc :leave (fn [state] (xiana/ok ((:leave in) state))))
+    (:error in) (assoc :error (fn [state] (xiana/error ((:error in) state))))))
+
+(defn middleware->enter
+  "Add middleware as the :enter function of interceptor."
+  ([middleware]
+   (middleware->enter {} middleware))
+  ([interceptor middleware]
+   (assoc interceptor :enter
+          (fn [{request :request :as state}]
+            (xiana/ok
+             (assoc state :request ((middleware identity) request)))))))
```

Comment:
Something like this?

```clojure
(-> interceptor
   (assoc :leave
          (fn [{response :response :as state}]
            (-> state
                (assoc :response ((middleware identity) response))
                (xiana/ok)))))))
```

Reply:
That looks better

Reply:
See https://github.com/Flexiana/framework/pull/88/commits/3268d767a68b1da13955189ce1ef7f1c19f5d062

Reply:
@lambdart, from what I have read, I think that this is a great attempt for making the framework's source code more readable.

If a absolutely had to comment about something, that would be the way the test preconditions are set. In both [Frankie](https://github.com/Flexiana/Frankie/blob/15fb05ce72d347c8fa5d73175b7ac83609562511/test/frankie_fixture.clj) and [framework](https://github.com/Flexiana/framework/blob/9958abec990b0e0b0a686948e350969d48dd0af9/test/framework_fixture.clj) we used to have fixtures for this. With fixtures, we can run the tests just by executing `lein run`. It is also easier to understand what happens there, since they are written in Clojure, compared to the [script/auto.sh](https://github.com/Flexiana/framework/blob/27d4840055ed8aac54cb973c9504c5ff301d9740/script/auto.sh) file.